uk_geo_utils/management/commands/import_onspd.py

```diff
@@ -55,25 +45,23 @@ def add_arguments(self, parser):
             choices=["may2018", "aug2022"],
         )
 
-    def handle(self, *args, **kwargs):
-        self.table_name = get_onspd_model()._meta.db_table
-        self.path = kwargs["path"]
-        self.header = HEADERS[kwargs.get("header", "aug2022")]
-        if kwargs["transaction"]:
-            with transaction.atomic():
-                self.import_onspd()
-        else:
-            self.import_onspd()
-
-    def import_onspd(self):
-        glob_str = os.path.join(self.path, "*.csv")
+    def get_table_name(self):
+        return get_onspd_model()._meta.db_table
+
+    def import_data_to_temp_table(self):
+        self.import_onspd(self.temp_table_name)
+
+    def import_onspd(self, table_name):
+        glob_str = os.path.join(self.data_path, "*.csv")
         files = glob.glob(glob_str)
         if not files:
-            raise FileNotFoundError("No CSV files found in %s" % (self.path))
+            raise FileNotFoundError(
+                "No CSV files found in %s" % (self.data_path)
+            )
 
         cursor = connection.cursor()
         self.stdout.write("clearing existing data..")
-        cursor.execute("TRUNCATE TABLE %s;" % (self.table_name))
+        cursor.execute("TRUNCATE TABLE %s;" % (table_name))
```

Comment:
If we're importing into a temp table, I think we can remove the `TRUNCATE` statement here.

Looks like we removed this in the addressbase variant.